azurerm/resource_arm_management_lock.go

```diff
@@ -0,0 +1,155 @@
+package azurerm
+
+import (
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/Azure/azure-sdk-for-go/arm/resources/locks"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceArmManagementLock() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmManagementLockCreateUpdate,
+		Read:   resourceArmManagementLockRead,
+		Delete: resourceArmManagementLockDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				// TODO: validation
```

Comment:
Do we want to add validation with this PR?

Reply:
done